python/lsst/display/firefly/firefly.py

```diff
@@ -165,7 +165,7 @@ def _mtv(self, image, mask=None, wcs=None, title=""):
             self._erase()
 
             with tempfile.NamedTemporaryFile() as fd:
-                afwDisplay.writeFitsImage(fd.name, image, wcs, title)
+                afwDisplay.writeFitsImage(fd.name, image, wcs, title, metadata=metadata)
                 fd.flush()
                 fd.seek(0, 0)
                 self._fireflyFitsID = _fireflyClient.upload_data(fd, 'FITS')
```

Comment:
For debugging purposes, I can confirm that something is being returned here, which means `status_code` was 200 based on this line: https://github.com/Caltech-IPAC/firefly_client/blob/016432396fd3e84475fc26b9538b77622f0df286/firefly_client/firefly_client.py#L563-L565

Reply:
Modifying this to `self._fireflyFitsID = _fireflyClient.upload_file(fd.name)` works. Similarly for the mask below.

Reply:
Thanks. This is a bug introduced in the move to python since the original code accepted a file descriptor and I thought I was being given a file name here. I should be able to support a file descriptor directly. I will have a look.